eras/conway/impl/test/Test/Cardano/Ledger/Conway/GovSpec.hs

```diff
@@ -6,15 +6,56 @@ module Test.Cardano.Ledger.Conway.GovSpec where
 import Cardano.Ledger.BaseTypes
 import Cardano.Ledger.Conway
 import Cardano.Ledger.Conway.Governance
+import Cardano.Ledger.Conway.PParams (ppGovActionLifetimeL)
+import Cardano.Ledger.Credential (Credential (KeyHashObj))
 import Cardano.Ledger.Shelley.LedgerState
+import Control.Monad (void)
 import Data.Default.Class (Default (..))
 import Data.Maybe
 import Lens.Micro
 import Test.Cardano.Ledger.Common
 import Test.Cardano.Ledger.Conway.ImpTest
 
 spec :: Spec
-spec =
+spec = do
+  describe "Votes fail as expected" $ do
+    itM @Conway "on expired gov-actions" $ do
+      modifyPParams $ ppGovActionLifetimeL .~ 2 -- Voting after the 3rd epoch should fail
+      khDRep <- setupSingleDRep
+      constitutionHash <- freshSafeHash
+      let constitutionAction =
+            NewConstitution
+              SNothing
+              ( Constitution
+                  ( Anchor
+                      (fromJust $ textToUrl "constitution.0")
+                      constitutionHash
+                  )
+                  SNothing
+              )
+      gaidConstitutionProp <- submitProposal constitutionAction
+      passEpoch
+      passEpoch
+      passEpoch
+      void $ voteForProposalFail (DRepVoter $ KeyHashObj khDRep) gaidConstitutionProp
```

Comment:
Could you figure out a way to also specify an expected predicate failure?

You'll probably need to improve or create an alternative version o `submitFailingTx`

Reply:
:100: :raised_hands: